### ISSUE TITLE
Allows to parse successfully some invalid dates

### ISSUE THREAD

**Issue by Allactaga** [dateutil/dateutil#31]
This probably should not be parsed at all:
```python
>>> from dateutil import parser
>>> parser.parse("2015 01 02 03 04 05 06 07 08 09")
datetime.datetime(2015, 1, 2, 9, 0)
```

**Comment by pganssle** [comment]
This happens because of the way that the dates are inferred from the tokens. It interprets the first 3 numbers as Year, Month, Date, then it has a bunch of 2-digit numeric tokens left. The logic is that if it finds a 2-digit numeric token *after* the year, month and date are determined, then that is an hour. As a result, each subsequent 2-digit number it finds just replaces the existing hour.

PR #32 fixes this, though it does so just by making sure that we've already found an hour, we won't continue to interpret this as the correct hour. Again, I'm thinking that this should be addressed in a more fundamental way when Issues #19 and #28 are addressed.

**Comment by pganssle** [comment]
Closed by #32.
